docs/architecture/adr-071-crypto-v2-multi-curve.md

```diff
@@ -0,0 +1,804 @@
+
+
+### Backwards Compatibility
+
+The proposed migration path is similar to what the cosmos-sdk has done in the past. To ensure a smooth transition, the following steps will be taken:
+
+* Develop all new code in [cosmos/crypto](https://github.com/cosmos/crypto) repo. Create unit tests, documentation and examples.
+* Deprecate the old crypto package. The old crypto package will still be usable, but it will be marked as deprecated and users can opt to use the new package.
+* Migrate the codebase to use the new cosmos/crypto package and remove the old crypto one.
+
+_A more detailed migration path is provided in the corresponding document._
+
+### Positive
+
+* Single place of truth
+* Easier to use interfaces
+* Easier to extend
+* Unit test for each crypto package
+* Greater maintainability
+* Incentivize addition of implementations instead of forks
+* Decoupling behavior from implementation
+* Sanitization of code
+
+### Negative
+
+* It will involve an effort to adapt existing code.
+* It will require attention to detail and audition.
+
+### Neutral
+
+* It will involve extensive testing.
+
+## Test Cases
+
+* The code will be unit tested to ensure a high code coverage
+* There should be integration tests around Keyring and CryptoProviders.
+
+## Further Discussions
+
+> While an ADR is in the DRAFT or PROPOSED stage, this section should contain a
+> summary of issues to be solved in future iterations (usually referencing comments
+> from a pull-request discussion).
+>
+> Later, this section can optionally list ideas or improvements the author or
+> reviewers found during the analysis of this ADR.
+
+
+## References
+
+* TPM 2.0 support: https://github.com/google/go-tpm
+* https://solanacookbook.com/references/keypairs-and-wallets.html#how-to-generate-a-new-keypair
+* https://www.nist.gov/news-events/news/2022/07/nist-announces-first-four-quantum-resistant-cryptographic-algorithms
+* https://blog.cloudflare.com/nist-post-quantum-surprise/
+* https://pkg.go.dev/crypto#Hash
+* https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html
+
+## Appendix
+
+
+### Implementation on CometBFT Codebase
+
+The implementation of this ADR in CometBFT will require a medium-level refactor of the codebase. Below we describe the proposed strategy to perform this implementation, which is:
+
+* Add a new `PrivValidator` implementation that uses the `CryptoProvider` interface underneath.
+
+* Adding [Keyring](https://github.com/cosmos/cosmos-sdk/blob/main/crypto/keyring/keyring.go) and [Record](https://github.com/cosmos/cosmos-sdk/blob/main/proto/cosmos/crypto/keyring/v1/record.proto) for storing and loading providers.
```

Comment:
Two comments here:
* Does the use of `CryptoProvider` make it inevitable that we use a keyring? Functionally, CometBFT just needs one and only one "record". I see this as overkill
* In the case we can't escape using a keyring, it can't be located in the SDK, as that will cause a dependency cycle

Reply:
Keyring and Record are optional for CometBFT (clarified that though). In the `Loading and storing implementations` section it is mentioned a 'greater impact' alternative vs not using Keyring, which I think is the easiest alternative for now.. 
Nevertheless most likely is that Keyring will be moved away from the sdk to the new cosmos/crypto repo, so the cyclic dependency wouldn't be an issue in that case.